06.bowling_object/game.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+class Game
+  LAST_FRAME = 9
+
+  def initialize(marks)
+    @frames = create_frames(marks)
+  end
+
+  def score
+    @frames.each_with_index.sum do |frame, index|
+      next frame.score if index == LAST_FRAME
+
+      if frame.strike?
+        frame.score + next_two_shots_score(index)
+      elsif frame.spare?
+        frame.score + next_shot_score(index)
+      else
+        frame.score
+      end
+    end
+  end
+
+  private
+
+  def create_frames(marks)
+    frames = []
+    index = 0
+
+    marks = marks.split(',')
+    LAST_FRAME.times do
+      if marks[index] == 'X'
+        frames << [marks[index]]
+        index += 1
+      else
+        frames << [marks[index], marks[index + 1]]
+        index += 2
+      end
```

Comment:
```suggestion
      length = marks[index] == 'X' ? 1 : 2
      frames << marks[index, length]
      index += length
```

ちょっと上級テクニックですがこんな感じで書けそうです

Reply:
その書き方は思いつきませんでした！変更しました。